lib/browser/growl.js

```diff
@@ -1,5 +1,207 @@
+      return Notification.requestPermission().then(function(permission) {
+        return permission === 'granted';
+      });
+    }
+  };
+
+  return permitted[Notification.permission]();
+}
+
+/**
+ * @summary
+ * Determines if notification should proceed.
+ *
+ * @description
+ * Notification shall <strong>not</strong> proceed unless `value` is true.
+ *
+ * `value` will equal one of:
+ * <ul>
+ *   <li><code>true</code> (from `isPermitted`)</li>
+ *   <li><code>false</code> (from `isPermitted`)</li>
+ *   <li><code>undefined</code> (from `Promise.race`)</li>
+ * </ul>
+ *
+ * @private
+ * @param {boolean|undefined} value - Determines if notification permissible.
+ * @returns {Promise<undefined>} Notification can proceed
+ */
+function canNotify(value) {
+  if (!value) {
+    var why = value === false ? 'blocked' : 'unacknowledged';
+    var reason = 'not permitted by user (' + why + ')';
+    return Promise.reject(new Error(reason));
+  }
+  return Promise.resolve();
+}
+
+/**
+ * Displays the notification.
+ *
+ * @private
+ * @param {Runner} runner - Runner instance.
+ */
+function display(runner) {
+  var stats = runner.stats;
+  var symbol = {
+    cross: '\u274C',
+    tick: '\u2705'
+  };
+  var logo = require('../../package').logo;
+  var lang = 'en-US';
+  var _message;
+  var message;
+  var title;
+
+  if (stats.failures) {
+    _message = stats.failures + ' of ' + runner.total + ' tests failed';
+    message = symbol.cross + ' ' + _message;
+    title = 'Failed';
+  } else {
+    _message = stats.passes + ' tests passed in ' + stats.duration + 'ms';
+    message = symbol.tick + ' ' + _message;
+    title = 'Passed';
+  }
+
+  // Send notification
+  var options = {
+    badge: logo,
+    body: message,
+    icon: logo,
+    lang: lang,
+    name: 'mocha',
+    requireInteraction: false,
+    tag: generateTag(),
+    timestamp: Date.now()
+  };
+  var notification = new Notification(title, options);
+
+  // Autoclose after brief delay (makes various browsers act same)
+  var FORCE_DURATION = 4000;
+  setTimeout(notification.close.bind(notification), FORCE_DURATION);
+}
+
+/**
+ * As notifications are tangential to our purpose, just log the error.
+ *
+ * @private
+ * @param {Error} err - Why notification didn't happen.
+ */
+function notPermitted(err) {
+  console.error('notification error:', err.message);
+}
+
+/**
+ * Used to help distinguish notifications within a root suite run.
+ * @type {number}
+ */
+var ntags = 0;
+
+/**
+ * @summary
+ * Generates a datestamped, run-unique tag for each notification.
+ *
+ * @description
+ * Use of a tag allows notifications that represent the same conceptual event
+ * to specify as much. This implementation doesn't really bother.
+ *
+ * @private
+ * @returns {string} tag unique within a root suite execution
+ * @example
+ *
+ * generateTag();
+ * // => "mocha_results-20181117-1"
+ */
+function generateTag() {
+  return ['mocha_results', datestamp(), ++ntags].join('-');
```

Comment:
if the implementation doesn't matter, why not just use `Date.now()` instead of the `datestamp()` impl below?

Reply:
Since based on msecs, it could _never_ be used to match up with previous notifications. This scheme allows for replacing previous notifications with newer ones.

Reply:
it sounds like then, the limitation is that only results that arrive on the same *day* can be replaced?  this is fine, but please mention the intent/reasoning in the comments, b/c it sounded like it didn't need to be anything in particular.  alternatively, just merge it and we'll fix it later.